src/Utility/AnaSol.C

```diff
@@ -107,8 +113,62 @@ template <class T> void parseDerivatives (T* func, const std::string& variables,
 AnaSol::AnaSol (const TiXmlElement* elem, bool scalarSol)
   : vecSol(nullptr), vecSecSol(nullptr), stressSol(nullptr)
 {
-  std::string variables;
+  const char* type = elem->Attribute("type");
+  if (type && !strcasecmp(type,"fields")) {
+    parseFieldFunctions(elem, scalarSol);
+    return;
+  }
+
+  parseExpressionFunctions(elem, scalarSol);
+}
+
+
+AnaSol::~AnaSol ()
+{
+  for (RealFunc* rf : scalSol)
+    delete rf;
+  for (VecFunc* vf : scalSecSol)
+    delete vf;
+  delete vecSol;
+  delete vecSecSol;
+  delete stressSol;
+}
+
+
+void AnaSol::initPatch(size_t pIdx)
+{
+  for (RealFunc* rf : scalSol) {
+    FieldFunction* ff = dynamic_cast<FieldFunction*>(rf);
+    if (ff)
+      ff->initPatch(pIdx);
+  }
+
+  for (VecFunc* rf : scalSecSol) {
+    VecFieldFunction* ff = dynamic_cast<VecFieldFunction*>(rf);
+    if (ff)
+      ff->initPatch(pIdx);
+  }
+  if (vecSol) {
+    VecFieldFunction* ff = dynamic_cast<VecFieldFunction*>(vecSol);
+    if (ff)
+      ff->initPatch(pIdx);
+  }
+  if (vecSecSol) {
+    TensorFieldFunction* ff = dynamic_cast<TensorFieldFunction*>(vecSecSol);
+    if (ff)
+      ff->initPatch(pIdx);
+  }
+  if (stressSol) {
+    STensorFieldFunction* ff = dynamic_cast<STensorFieldFunction*>(stressSol);
+    if (ff)
+      stressSol->initPatch(pIdx);
+  }
```

Comment:
Wouldn't it be nicer to have a `virtual void initPatch(size_t) {}` in the `FunctionBase` class instead, such that we don't need to to all this dynamic_cast-ing?

Reply:
sure. didn't know which solution you preferred.